apps/angular/src/core/services/anime.service.ts

```diff
@@ -9,26 +9,40 @@ import { Pagination } from '@js-camp/core/models/pagination';
 import { PaginationMapper } from '@js-camp/core/mappers/pagination.mapper';
 import { AnimeMapper } from '@js-camp/core/mappers/anime/anime.mapper';
 
+import { PaginationConfig } from '@js-camp/core/interfaces/pagination';
+
 import { ApiService } from './api.service';
+import { NavigateService } from './navigate.service';
 
 /** Api service. */
 @Injectable({
   providedIn: 'root',
 })
-
 export class AnimeService {
-  public constructor(private apiService: ApiService) {}
-
-  private formatErrors(error: string): Error {
-    return new Error(error);
-  }
+  public constructor(
+    private readonly apiService: ApiService,
+    private readonly navigateService: NavigateService,
+  ) {}
 
   /**
-   * Fetch anime from server.
-   * @param params Parameters for request.
-   **/
-  public fetchAnime(params?: HttpParams): Observable<Pagination<Anime>> {
-    const animeResponse$ = this.apiService.get<PaginationDto<AnimeDto>>(`anime/anime/`, params);
-    return animeResponse$.pipe(map(animes => PaginationMapper.fromDto(animes, animeDto => AnimeMapper.fromDto(animeDto))));
+   * Fetch anime data from server.
+   * @param config Configuration for request.
+   */
+  public fetchAnime(config: PaginationConfig): Observable<Pagination<Anime>> {
+    const path = 'anime/anime/';
+    const params = new HttpParams({
+      fromObject: {
+        ...PaginationMapper.toDto(config),
+      },
+    });
+    const animeResponse$ = this.apiService.get<PaginationDto<AnimeDto>>(
+      path,
+      params,
+    );
+    return animeResponse$.pipe(
+      map(animes =>
+        PaginationMapper.fromDto(animes, animeDto =>
+          AnimeMapper.fromDto(animeDto))),
+    );
```

Comment:
Do you need `animeResponse$` variable? Maybe return stream right away?
```suggestion
    return this.apiService.get<PaginationDto<AnimeDto>>(path, params).pipe(
      map(animes =>
        PaginationMapper.fromDto(animes, animeDto =>
          AnimeMapper.fromDto(animeDto))),
    );
```